File: html/old/R2/SACO R2 2009 (Solutions)/button.java
// SACO 2nd Round 2009
// Problem: Unmarked calculator button
// Author:  Robert Ketteringham

import java.io.*;
class button {
    public static void main (String [] args) throws IOException {
        BufferedReader kb = new BufferedReader (new InputStreamReader(System.in));
        System.out.print("First Number: ");
        int a = Integer.parseInt(kb.readLine());
        System.out.print("Second Number: ");
        int b = Integer.parseInt(kb.readLine());
        System.out.print("Answer: ");
        int answer = Integer.parseInt(kb.readLine());
        // Deciding  whether it is plus or times. Note that nested "if's" are also possible
        if (a + b == answer && a * b == answer)
            System.out.println("Either Plus or Times");
        else if (a + b == answer)
            System.out.println("Plus Only");
        else if (a * b == answer)
            System.out.println("Times Only");
        else
            System.out.println("Neither Plus nor Times");
        // Printing the two sums
        System.out.println(a + "+" + b + "=" + (a+b));
        System.out.println(a + "x" + b + "=" + (a*b));
    }
}

File: html/old/backup/sacor2_2009/button.py
#!/usr/bin/env python
# SACO Second Round 2009
# Problem: Unmarked Calculator Button
# Author:  Michiel Baird

a = int(raw_input("First Number: "))
b = int(raw_input("Second Number: "))
c = int(raw_input("Answer: "))
if a+b == c:
    if a*b == c:
        print "Either Plus or Times"
    else:
        print "Plus only"
elif a*b == c:
    print "Times only"
else:
    print "Neither Plus nor Times"
print "%d+%d=%d" % (a,b,a+b)
print "%dx%d=%d" % (a,b,a*b)
